element_interface/intan_loader.py

```diff
@@ -248,40 +240,56 @@ def load_rhs(folder: str, file_expr: str):
             rhs_data["timestamps"] (np.array_like): Relative timestamps in seconds.
     """
 
+    rhs_data = {}
+
+    # Get header
     header_filepath = next(Path(folder).glob("info.rhs"))
     with open(header_filepath, "rb") as fid:
-        header = read_header(fid)
+        rhs_data["header"] = _read_header(fid)
 
+    # Get timestamps
     time_file = next(Path(folder).glob("time.dat"))
 
-    timestamps = (
+    rhs_data["timestamps"] = (
         np.memmap(time_file, dtype=np.int32)
-        / header["frequency_parameters"]["amplifier_sample_rate"]
+        / rhs_data["header"]["frequency_parameters"]["amplifier_sample_rate"]
     )
 
-    rhs_data = dict(header=header, timestamps=timestamps, recordings={})
-
+    # Get data files
     file_paths = Path(folder).glob(file_expr)
-    file_paths = [x for x in file_paths if x.as_posix() != "time.dat"]
 
-    for file_path in file_paths:
-        file_path = file_path.as_posix()
-        if "amp" in file_path:
+    if file_expr == "*":
+        exclude_list = ["time", "info", "Zone.Identifier"]
+
+        file_paths = [
+            file
+            for file in file_paths
+            if not any([string in file.as_posix() for string in exclude_list])
+        ]
+
+    # Get recording data
+    rhs_data["recordings"] = {}
+
+    for file_path in sorted(file_paths):
+        signal_type = file_path.stem.split("-")[0]
+
+        if signal_type == "amp":
             signal = np.memmap(file_path, dtype=np.int16)
             signal = signal * 0.195  # Convert to microvolts
-        elif "board-ANALOG-IN" in file_path or "board-ANALOG-OUT" in file_path:
+            
+        elif signal_type == "board":
             signal = np.memmap(file_path, dtype=np.uint16)
             signal = (signal - 32768) * 0.0003125  # Convert to volts
-        elif "dc-" in file_path:
+
+        elif signal_type == "dc":
             signal = np.memmap(file_path, dtype=np.uint16)
             signal = (signal - 512) * 19.23  # Convert to milivolts
-        elif "board-DIGITAL-IN" in file_path or "board-DIGITAL-OUT" in file_path:
+
+        elif signal_type == "stim":
             signal = np.memmap(file_path, dtype=np.uint16)
-        elif "stim-" in file_path:
-            data = np.memmap(file_path, dtype=np.uint16)
-            i = np.bitwise_and(data, 255) * header["stim_step_size"]
-            sign = (128 - np.bitwise_and(data, 255)) / 128
+            i = np.bitwise_and(signal, 255) * rhs_data["header"]["stim_step_size"]
+            sign = (128 - np.bitwise_and(signal, 255)) / 128
             signal = i * sign
```

Comment:
```suggestion
            current = np.bitwise_and(signal, 256) * rhs_data["header"]["stim_step_size"]
            sign = (128 - np.bitwise_and(signal, 256)) / 128
            signal = current * sign
```

Reply:
@dimitri-yatsenko How about now?

Reply:
`current` is a bit confusing since we are dealing with ephys.

Reply:
It is directly from the matlab code.

```
stimulation output current
 if allchannelsstimulatedsimultaneously
     stimchvect = 1; %all channels are electrically stimulated at the same time with the same waveform, so need just one channel for timing
 else
     stimchvect = 1 : numel(data.amplifier_channels);
 end
 for ch = stimchvect
     filename = sprintf('stim-%s.dat', data.amplifier_channels(ch).native_channel_name);
     fileinfo = dir(filename);
     if ~isempty(fileinfo)
         fprintf('   loading stimulated data for channel %d (%s)\n', ch, filename);
         num_samples = fileinfo.bytes/2; % uint16 = 2 bytes
         fid = fopen(filename, 'r');
         dat = fread(fid, num_samples, 'uint16');
         fclose(fid);
         curmag = bitand(dat, 255) * data.stim_parameters.stim_step_size; % current magnitude
         signat = (128 - bitand(dat, 256))/128; % convert sign bit to 1 or -1
         data.stim_data(ch, :) = curmag .* signat; % signed current in Amps
     end
 end
```

Reply:
I am not quite sure what's happening here. Why are we getting the 9th bit? Still does not seem to make sense.

Reply:
It makes sense now. There were errors in both versions mixing up the constants 255 and 256.  Please fix them and I'll accept.

Reply:
@JaerongA The suggestion above will fix the bug.

Reply:
Please add an explanation in the comment. It's weird that they are using the 9-bit one's complement for signed numbers.  Two's complement has been the standard in computing. The comment could be something like:

```python
# convert the signal stored in 9-bit one's complement to standard encoding
```

Reply:
Thank, @tdincer @dimitri-yatsenko. Suggestions have been committed now.